rust/parquet/src/basic.rs

```diff
@@ -156,6 +158,12 @@ pub enum ConvertedType {
 // Mirrors `parquet::LogicalType`
 
 /// Logical types used by version 2 of the Parquet format.
+///
+/// *Upgrade Note*: This is an *entirely new* struct as of version
```

Comment:
I'm not sure if its useful to have this message here as well. Maybe the one in README is enough?

Reply:
I think it is important to put a hint as close to where people will hit the compiler error as possible (e.g. I wouldn't have known to look in the README when I hit the error on upgrade). 

That being said, I think you are right that the actual content doesn't need to be duplicated here -- I will change it to point people at the readme for further details.